File: src/lib/types/goal.ts
export type GoalType = 'daily-reading' | 'books-per-month' | 'books-per-year' | 'reading-streak' | 'pages-per-day';
export type GoalPeriod = 'day' | 'week' | 'month' | 'year';

export interface Goal {
  id: string;
  type: GoalType;
  target: number;
  period: GoalPeriod;
  startDate: Date;
  endDate?: Date;
  active: boolean;
  createdAt?: Date;
  updatedAt?: Date;
}

export interface CreateGoalInput {
  type: GoalType;
  target: number;
  period: GoalPeriod;
  startDate?: Date;
  endDate?: Date;
}

export interface UpdateGoalInput {
  target?: number;
  endDate?: Date;
  active?: boolean;
}

File: ios/RaquelReads/RaquelReads/Models/Goal.swift
import Foundation
import SwiftData

@Model
final class Goal {
    @Attribute(.unique) var id: UUID
    var type: GoalType
    var target: Int
    var period: GoalPeriod
    var startDate: Date
    var endDate: Date?
    var active: Bool
    var createdAt: Date
    var updatedAt: Date

    init(
        id: UUID = UUID(),
        type: GoalType,
        target: Int,
        period: GoalPeriod,
        startDate: Date = Date(),
        endDate: Date? = nil,
        active: Bool = true,
        createdAt: Date = Date(),
        updatedAt: Date = Date()
    ) {
        self.id = id
        self.type = type
        self.target = target
        self.period = period
        self.startDate = startDate
        self.endDate = endDate
        self.active = active
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
}
